python/fusion_engine_client/analysis/data_loader.py

```diff
@@ -739,24 +741,84 @@ def _to_gps_or_p1(value):
                 time_sec = np.array((_to_gps_or_p1(times),))
 
         # Now, find all values that are GPS time (i.e., big enough that we assume they're not P1 times already) and
-        # convert them to P1 time.
-        gps_idx = is_gps_time(time_sec)
-        if np.any(gps_idx):
-            if p1_ref_sec is None:
-                time_sec[gps_idx] = np.nan
-            else:
-                # The relationship between P1 time and GPS time should not change rapidly since P1 time is rate-steered
-                # to align with GPS time. As a result, it should be safe to extrapolate between gaps in P1 or GPS times,
-                # as long as the gaps aren't extremely large. NumPy's interp() function does not extrapolate, so we
-                # instead use SciPy's function.
-                f = sp.interpolate.interp1d(gps_ref_sec, p1_ref_sec, fill_value='extrapolate')
-                time_sec[gps_idx] = f(time_sec[gps_idx])
+        # convert them to P1 time or vice versa.
+        if gps_to_p1:
+            gps_idx = is_gps_time(time_sec)
+            if np.any(gps_idx):
+                if p1_ref_sec is None:
+                    time_sec[gps_idx] = np.nan
+                else:
+                    # The relationship between P1 time and GPS time should not change rapidly since P1 time is rate-steered
+                    # to align with GPS time. As a result, it should be safe to extrapolate between gaps in P1 or GPS times,
+                    # as long as the gaps aren't extremely large. NumPy's interp() function does not extrapolate, so we
+                    # instead use SciPy's function.
+                    f = sp.interpolate.interp1d(gps_ref_sec, p1_ref_sec, fill_value='extrapolate')
+                    time_sec[gps_idx] = f(time_sec[gps_idx])
+        else:
+            p1_idx = not is_gps_time(time_sec)
```

Comment:
`is_gps_time()` returns a boolean numpy array. `not` won't work on a logical array, you need to use `~`.

Reply:
Ah, this actually isn't quite right. It can return a bool or a numpy array. In my script it returns a bool so `~` actually won't work.